src/patternfly/components/JumpLinks/jump-links.scss

```diff
@@ -98,6 +101,18 @@
     border: solid var(--pf-c-jump-links__list--before--BorderColor);
     border-width: var(--pf-c-jump-links__list--before--BorderTopWidth) var(--pf-c-jump-links__list--before--BorderRightWidth) var(--pf-c-jump-links__list--before--BorderBottomWidth) var(--pf-c-jump-links__list--before--BorderLeftWidth);
   }
+
+  .pf-c-jump-links__list {
+    --pf-c-jump-links__list--PaddingTop: 0;
```

Comment:
You could add `--pf-c-jump-links__list--before--BorderColor: transparent;` here to disable border from nested lists, though it isn't necessary since it's the same color and overlaps with the parent's border.